src/cdk/schematics/update-tool/component-resource-collector.ts

```diff
@@ -133,13 +133,11 @@ export class ComponentResourceCollector {
         property.initializer.elements.forEach(el => {
           if (ts.isStringLiteralLike(el)) {
             const stylesheetPath = this._fileSystem.resolve(sourceFileDirPath, el.text);
+            const stylesheet = this.resolveExternalStylesheet(stylesheetPath, node);
 
-            // In case the stylesheet does not exist in the file system, skip it gracefully.
-            if (!this._fileSystem.exists(stylesheetPath)) {
-              return;
+            if (stylesheet) {
```

Comment:
The change in general is good (enabling strict), but I wonder how this issue surfaces? Do we understand how this can happen? We asserted that a file exists in the virtual tree, and then read from it? Reading the file from the tree should not be `null`, or is this an edge case when a file is not readable somehow?

https://cs.opensource.google/angular/angular-cli/+/master:packages/angular_devkit/schematics/src/tree/host-tree.ts;l=282-291?q=read&ss=angular%2Fangular-cli

Reply:
The assertion here was fine, but I think it failed [on the one below](https://github.com/angular/components/pull/19783/files#diff-7e114be5b3b61d197fa40462f633cb0bR176) where we didn't have an assertion.

Reply:
Unless I miss something, we had a check before we invoked `resolveExternalStylesheet`. And the other one just accepted a glob of source files which are guaranteed to exist in FS.

Maybe the glob returned source files which exist on FS but are not available in the virtual file system. That would seem most likely.

Reply:
I guess it depends on whether [the `resolve` call here](https://github.com/angular/components/blob/master/src/cdk/schematics/update-tool/index.ts#L106) corresponds to a `path.resolve` from Node which won't check the file system, or whether it's something different that actually does the lookup. If it's the latter, your explanation sounds plausible. Either way, I think that we should get this in for the next release so people that are updating to v10 don't run into it.

Reply:
The `resolve` call should not be relevant here. The list of `additionalStylesheetPaths` is based on a real file system query, and `resolve` just turns the absolute FS path into a virtual FS workspace relative path.

So there are two options: Either the files are somehow not readable and are not added to the virtual FS tree, or our path resolution logic is flawed. And this is the point/issue I think we should figure out. Fixing the issue is great, but understanding the root cause seems beneficial. To be clear: From the beginning on I agreed that this is a good change to land either way, but it would be still worth figuring this out I assume. I'll ask the issue reporter for more information.

Also I think since this has been reported only once and seems very edgy/surprising, this is probably not P2 worthy.